### ISSUE TITLE
Different frame outputs across images/video and executables

### ISSUE THREAD

**Issue by d-bohn** [TadasBaltrusaitis/OpenFace#892]
I'm using the official Dockerfile that I built locally on my machine. I have a video with 5395 frames and multiple faces which I pass to OpenFace with the following command line arguments
```
FaceLandMarkVidMulti \
    -f path/to/video.avi
    -out_dir path/to/of_vid_multi
```
I would expect that the output `path/to/of_vid_multi/video.csv` to have a max value of 5395 in the `frame` column. However, what I get is 5390. Weird. So I re-run the analysis with
```
FeatureExtraction \
    -f path/to/video.avi
    -out_dir path/to/of_vid_multi2
```
Now when I check `path/to/of_vid_multi2/video.csv` I get the expected max value of 5395 frames, but it only captures the largest face as to be expected.

I thought it might be that the video can't be encoded/decoded/read properly with ffmpeg, so I made an image stack of frames from the video. I pass all 5395 images to OpenFace with the command line arguments
```
FaceLandMarkVidMulti \
    -fdir path/to/frames
    -out_dir path/to/of_frame_multi
```
Inspecting the frame analysis at `path/to/of_frame_multi/frames.csv` _should_ now have all 5395 frames analyzed, but upon inspection of that file the max value in the `.csv` is 5366!

Again I re-do this analysis by passing the frames to  `FeatureExtraction` and I get the expected 5395 frames. I'm not sure why I would be seeing different values across videos and images and different executables. Any insights would be appreciated.

**Comment by TadasBaltrusaitis** [comment]
That's really unexpected behavior, especially as input for FeatureExtraction and FaceLandmarkVidMulti is done using the same utility functions, the non-deterministic behavior is especially worrying. There are instances when OpenCV (which is used for video reading) gets the wrong number of input frames compared to other tools, so when exact precision of frame processing is needed (e.g. when aligning with other modalities such as audio), I would recommend using the method of dumping the video into frames.

**Comment by d-bohn** [comment]
It certainly is curious and concerning. Note that I did do as you suggest (produce image frame stacks to pass to the OpenFace), but I still see the same issue as with the video. Do you have any idea why `FaceLandmarkVidMulti` would produce the expected output for the short video (11 frames) but not for the longer video (5395 frames)? I suspected that `FaceLandmarkVidMulti` might not produce output for frames where it failed to detect any faces, but with the short video there are 8 frames where there are no faces, so this appears to be related specifically to `FaceLandmarkVidMulti` and  video/image length.

**Comment by TadasBaltrusaitis** [comment]
Not sure what exactly is causing it, but I'll try to investigate.

**Comment by d-bohn** [comment]
I did a bit more experimenting and I found something peculiar: Recall that with my 11 frame standard video that both  `FeatureExtraction` and `FaceLandMarkVidMulti` produce output with 11 rows. However, when I extract the frames from a longer video (395 frames) and run it through `FaceLandMarkVidMulti` the output is close to the expected number of frames/images, but the odd thing I noticed is that the first observation starts with frame/image 9.

Is there a reason why in one case `FaceLandMarkVidMulti` would produce "empty" rows for frames/images where it fails to detect any face, but in another just omit those cases entirely?

**Comment by TadasBaltrusaitis** [comment]
You're absolutely right, I just had another look at the code: `FaceLandmarkVidMulti`, will not report about frames where there are no detected faces, the recording of information is done only when there are face, while `FeatureExtraction` will, even if there are no faces in the scene.

However, the frame numbers should still be correct, even if the row numbers are not consistent.
